src/box/tuple_format.c

```diff
@@ -1187,7 +1193,7 @@ tuple_field_map_create_plain(struct tuple_format *format, const char *tuple,
 							     next_pos) != 0)
 				goto error;
 		}
-		if (validate)
+		if (validate_required_fields)
 			bit_clear(required_fields, field->id);
 		if (field->offset_slot != TUPLE_OFFSET_SLOT_NIL &&
 		    field_map_builder_set_slot(builder, field->offset_slot,
```

Comment:
What about dropping the field map builder from this function?

Reply:
It doesn't affect performance, according to profiling.